src/didactUriCompletionItemProvider.ts

```diff
@@ -18,271 +18,172 @@
 'use strict';
 
 import * as vscode from 'vscode';
-import * as extensionFunctions from './extensionFunctions';
-import {DidactUri} from './didactUri';
+import * as path from 'path';
+import * as fs from 'fs';
 
+const DIDACT = 'didact';
+const COMMAND_ID = 'commandId';
 
-export const didactProtocol = 'didact://?';
+export const didactProtocol = `${DIDACT}://?`;
+export const DIDACT_COMMAND_PREFIX = `${didactProtocol}${COMMAND_ID}=`;
 
 export class DidactUriCompletionItemProvider implements vscode.CompletionItemProvider {
 	
 	private extContext!: vscode.ExtensionContext;
+	private completionCatalog: any;
 
 	public constructor(ctxt: vscode.ExtensionContext) {
 		this.extContext = ctxt;
+		this.completionCatalog = this.getCompletionCatalog(this.extContext);
 	}
 
-	provideCompletionItems(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken, context: vscode.CompletionContext): vscode.ProviderResult<vscode.CompletionItem[] | vscode.CompletionList> {
-		const lineToSearch = document.lineAt(position).text;
-		if (lineToSearch.indexOf(didactProtocol) > -1) {
-			return this.provideCompletionItemsForDidactProtocol(lineToSearch);
-		}
-		return this.provideCompletionItemsOutsideDidactURI(lineToSearch);
-	}
-
-	protected provideCompletionItemsOutsideDidactURI(lineToSearch: string) : vscode.CompletionItem[] {
-		const completions: vscode.CompletionItem[] = [];
-
-		if (lineToSearch.indexOf(didactProtocol) === -1) {
-			this.insertRedhatDidactLinkCompletion("Insert link to start Didact from File Elsewhere in Extension Folder", completions);
-			this.insertDidactProtocolStarterCompletion("Start a new Didact link", completions);
-		}
-
-		return completions;
+	// public for testing
+	public getCompletionCatalog(context: vscode.ExtensionContext) : JSON {
+		const uri : vscode.Uri = vscode.Uri.file(
+			path.join(context.extensionPath, 'src/didactCompletionCatalog.json')
```

Comment:
The `src` folder is not accessible in the bundled `vsix` https://github.com/redhat-developer/vscode-didact/blob/f3d801f5c7e2cf5251ebdf889e52938fcc153146/.vscodeignore#L4
it needs to be accessed in another way. Or stored in another specific folder. Maybe in the `resources` fo ran existing one.